orttraining/orttraining/python/training/ortmodule/ortmodule.py

```diff
@@ -68,79 +60,78 @@ def forward(self, *inputs, **kwargs):
     def _apply(self, fn):
         """Override original method to delegate execution to the flattened PyTorch user module"""
 
-        # Delegation must happen to _flattened_module since methods depend on
-        # _apply to recursively apply the internal setting changes
-        self._module_metadata.flattened_module._apply(fn)
+        self._module_manager._apply(fn)
         return self
 
     def apply(self: T, fn: Callable[['Module'], None]) -> T:
         """Override original method to delegate execution to the flattened PyTorch user module"""
 
-        # Delegation must happen to _flattened_module since methods depend on
-        # apply to recursively apply the internal setting changes
-        self._module_metadata.flattened_module.apply(fn)
+        self._module_manager.apply(fn)
         return self
 
     def _is_training(self):
-        return self.training and torch.is_grad_enabled()
+        return self._module_manager.is_training()
 
     def train(self: T, mode: bool = True) -> T:
```

Comment:
Maybe pushing all Torch Modules APIs to the bottom of ORTModule class and add a header similar to

```python
################################################################################
# The methods below are part of torch.nn.Module API that are encapsulated through TorchModuleInterface
################################################################################
```

might be intuitive and allow devs to skip that part